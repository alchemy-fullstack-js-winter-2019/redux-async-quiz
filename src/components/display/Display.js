import React from 'react';
import PropTypes from 'prop-types';

export default class Display extends React.PureComponent {
    static propTypes = {
      image: PropTypes.string,
      fetch: PropTypes.func.isRequired,
    };

    componentDidMount() {
      this.props.fetch();
    }

    render() {
      return (
            <>
                <h1> hello from display comp</h1>
                <button onClick={this.props.fetch}>GET NEW DAWG</button>
                <img src={this.props.image}/>
                
            </>
      );
    }
}
