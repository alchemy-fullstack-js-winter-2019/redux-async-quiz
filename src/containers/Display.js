import { connect } from 'react-redux';
import Display from '../components/display/Display.js';
import { getDogAction } from '../actions/Dog.js';
import { getDog } from '../selectors/Dog.js';

const mapStateToProps = (state)=> ({
  image: getDog(state)
});

const mapDispatchToProps = dispatch => ({
  fetch() {
    dispatch(getDogAction());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(Display);


