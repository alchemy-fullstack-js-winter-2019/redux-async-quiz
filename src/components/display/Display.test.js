import React from 'react';
import renderer from 'react-test-renderer';
import Display from './Display';

describe('display component tests', () => {
  it('renders the display component correctly', () => {
    const fetch = jest.fn();
    const tree = renderer.create(
      <Display
        fetch={fetch}
        image='something'
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
