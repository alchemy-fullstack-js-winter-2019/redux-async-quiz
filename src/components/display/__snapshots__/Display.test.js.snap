// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display component tests renders the display component correctly 1`] = `
Array [
  <h1>
     hello from display comp
  </h1>,
  <button
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  >
    GET NEW DAWG
  </button>,
  <img
    src="something"
  />,
]
`;
